src/resource.js

```diff
@@ -1,6 +1,58 @@
 'use strict'
 
-export default () => {
+import url from 'url'
+import jts from 'jsontableschema'
+import _ from 'lodash'
 
-}
+export default class Resource {
 
+  constructor(descriptor) {
+    this._descriptor = descriptor
+    this.REMOTE_PROTOCOLS = ['http:', 'https:', 'ftp:', 'ftps:']
+  }
+
+  getDescriptor() {
+    return this._descriptor
+  }
+
+  getName() {
+    return this.getDescriptor()['name']
+  }
+
+  getType() {
+    const resourceField = this._getResourceField()
+
+    if (resourceField === 'data') {
+      return 'inline'
+    }
+
+    const protocol = url.parse(this.getDescriptor()[resourceField])['protocol']
+    if (_.includes(this.REMOTE_PROTOCOLS, protocol)) {
+      return 'remote'
+    } else {
+      return 'local'
+    }
+
+  }
+
+  getSource() {
+    return this.getDescriptor()[this._getResourceField()]
+  }
+
+  getTable() {
+    const resourceField = this._getResourceField()
+    return new jts.Table(this.getDescriptor()['schema'],
+                         this.getDescriptor()[resourceField])
+  }
+
+  _getResourceField() {
```

Comment:
You are right about the naming, it's ambiguous and I will rename it.

About why this is outside of `resource.(get)type` - it's because this logic is also used in resource.(get)source so I wanted to reuse the code.